### PR TITLE
Make sure we deal all options for each output class.

### PR DESCRIPTION
**Pull request by MarcCote**
For instance in data_understanding we have "a|A", "b|B", etc. We need to shift logits for all the possible options.

### CODE REVIEW

**Comment by MarcCote** [comment]
@sordonia, thoughts?

**Review by matheper — APPROVED** [review]
LGTM!

**Comment by MarcCote** [comment]
Let's merge it. Then
